include/boost/windows_sspi/detail/sspi_impl.hpp

```diff
@@ -203,20 +361,36 @@ class sspi_decrypt {
   // TODO: Make private
   std::vector<char> encrypted_data;
   std::vector<char> decrypted_data;
-  SECURITY_STATUS last_error;
+
+  boost::system::error_code last_error() const {
+    return error::make_error_code(m_last_error);
+  }
 
 private:
   CtxtHandle* m_context;
+  SECURITY_STATUS m_last_error;
 };
 
 class sspi_impl {
 public:
-  sspi_impl(CtxtHandle* context)
-    : encrypt(context)
-    , decrypt(context) {
+  sspi_impl(CredHandle* cred_handle)
+    : m_cred_handle(cred_handle)
+    , handshake(&m_context, cred_handle)
+    , encrypt(&m_context)
+    , decrypt(&m_context) {
+  }
+
+  ~sspi_impl() {
```

Comment:
This class controls a resource and is (I think) copyable - you should probably disable the copy constructor and assignment operator.

Reply:
Whoops.

This should definitely not be copyable as the CtxHandle keeps some state related to the current connection/stream, so even if it is possible to implement a copy construction/assignment, it most likely wouldn't make any sense.

Nicely spotted, thanks.